src/Faker/Provider/Miscellaneous.php

```diff
@@ -318,6 +318,8 @@ public static function currencyCode()
      */
     public static function emoji()
     {
-        return static::randomElement(static::$emoji);
+        $randomElement = static::randomElement(static::$emoji);
+        $unicodeHex = preg_replace("/\\\uD83D\\\uDE([0-9A-F]*)/", "&#x1F6\\1;", $randomElement);
```

Comment:
Doesn't this add a dependency on `ext-pcre`, or is this library non-optional when building PHP?

Reply:
Based on this page, it is bundled with PHP, but perhaps not enabled by default. <https://www.php.net/manual/en/extensions.membership.php>

Reply:
So, no better than `ext-json`?

Reply:
Technically it's the same, practically I've never found pcre to be disabled after an install except for on FreeBSD (where you basically have to explicitly install every non-core extension).

Reply:
`preg_replace` we use in a couple of places, `json_decode` only here.

Reply:
This seems very complicated. Let's just ignore this in BC checker.